src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/magical/MagicalZombiePills.java

```diff
@@ -56,11 +61,11 @@ public EntityInteractHandler getItemHandler() {
 
                 p.playSound(p.getLocation(), Sound.ENTITY_ZOMBIE_VILLAGER_CONVERTED, 1, 1);
 
-                ZombieVillager zombieVillager = (ZombieVillager) entity;
-                zombieVillager.setConversionTime(1);
-
-                if (SlimefunPlugin.getMinecraftVersion().isAtLeast(MinecraftVersion.MINECRAFT_1_15)) {
-                    zombieVillager.setConversionPlayer(p);
+                if (entity instanceof ZombieVillager) {
+                    healZombieVillager((ZombieVillager) entity, p);
+                }
+                else {
```

Comment:
```suggestion
                else if (SlimefunPlugin.getMinecraftVersion().isAtLeast(MinecraftVersion.MINECRAFT_1_16) && entity instanceof PigZombie) {
```

Reply:
I did the minecraft version check above it, is that not good?